packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -372,12 +372,19 @@ export default function VisualEditor( { styles } ) {
 							! isTemplateMode && (
 								<>
 									<LayoutStyle
-										selector=".edit-post-visual-editor__post-title-wrapper, .block-editor-block-list__layout.is-root-container"
+										selector=".edit-post-visual-editor__post-title-wrapper"
 										layout={ fallbackLayout }
 										layoutDefinitions={
 											globalLayoutSettings?.definitions
 										}
 									/>
+									<LayoutStyle
+										selector=".block-editor-block-list__layout.is-root-container"
+										layout={ blockListLayout }
+										layoutDefinitions={
+											globalLayoutSettings?.definitions
+										}
+									/>
```

Comment:
I'm no longer sure because this code has gone through a few changes 😅 but I think the first instance adds the initial computed layout for classic and block themes, and the second one only applies to block themes and adds any changes to Post Content if the template is edited.